N24DCCN162/n24dccn162_buoi5/bai_10.cpp

```diff
@@ -0,0 +1,33 @@
+#include <iostream>
+#include <cmath>
+using namespace std;
+int main() {
+    int n;
+    cout << "Hay nhap 1 so: ";
+    cin >> n;
+    if (n <= 1) {
+        cout << "Khong co phan tich thua so nguyen to cho n <= 1." << endl;
+        return 0;
+    }cout << n << " = ";
+    bool so=true;
+    for (int i=2; i<= sqrt(n);i++) {

```

Comment:
Không khuyến khích sử dụng vòng for với n bị thay đổi trong quá trình lặp (Vì vòng for là vòng lặp biết trước số lần lặp, với n thay đổi, số lần lặp của vòng for sẽ không biết trước, dẫn tới sai về ý nghĩa sử dụng, ở đây sài while sẽ hay hơn. Nhưng cách xử lý tối ưu, ok)